### PR TITLE
Drop pandas DataFrame NaN values for json protocol

### PR DESCRIPTION
**Pull request by tux-00**
This PR drop NaN values when converting df to dict in json protocol. I removed staticmethod decorator from _convert_dataframe_to_json to use a new static method.

Related to #422, #576, #436

### CODE REVIEW

File: influxdb/_dataframe_client.py [hunk 1]
```diff
@@ -255,14 +255,16 @@ def _convert_dataframe_to_json(dataframe,
...
+
         points = [
             {'measurement': measurement,
              'tags': dict(list(tag.items()) + list(tags.items())),
              'fields': rec,
              'time': np.int64(ts.value / precision_factor)}
-            for ts, tag, rec in zip(dataframe.index,
-                                    dataframe[tag_columns].to_dict('record'),
-                                    dataframe[field_columns].to_dict('record'))
+            for ts, tag, rec in dataframe_zip
```
Comment (Sonmer) [inline influxdb/_dataframe_client.py]: Got an error when all fields contain NaN values. I changed this to 
`for ts, tag, rec in dataframe_zip if any(rec)`
to skip the points with all empty fields.

**Comment by xginn8** [comment]
@tux-00 would you mind adding a test to check that NaNs are handled properly?

**Comment by tux-00** [comment]
Yes, when I have time to do it ! I keep this PR in mind.

**Comment by tux-00** [comment]
The test is OK with 2.7 and 3.6 but with 3.5 I got a strange error "Plugin 'travis' could not be loaded".

**Comment by tux-00** [comment]
Hey, can someone take a look at the CI error ? I need some help here.

**Comment by aviau** [comment]
Did you rebase on master recently? AFAIK some CI issues were fixed there.

**Comment by tux-00** [comment]
Is the rebase good ? Maybe I made something bad again.

**Comment by tux-00** [comment]
Can anyone merge this PR ?

**Comment by tux-00** [comment]
Is there someone here ?

**Comment by aviau** [comment]
I'll look at this when I have time, sorry for the delay. Maybe someone else from the team will have time before me.

**Comment by sebito91** [comment]
These changes were addressed in #812 , closing for now.

**Pull request closed without merging**